integration_test/third_party_apps_test.go

```diff
@@ -294,19 +421,15 @@ func runSingleTest(ctx context.Context, logger *logging.DirectoryLogger, vm *gce
 		}
 	}
 
-	metricName, err := findMetricName(app)
-	if err != nil {
-		return nonRetryable, fmt.Errorf("error finding metric name for %v: %v", app, err)
-	}
-	if metricName == "" {
-		logger.ToMainLog().Println("metric_name.txt is empty, skipping metrics testing...")
-		return nonRetryable, nil
-	}
-	// Assert that the right metric has been uploaded for the given instance
-	// at least once in the last hour.
-	if err = gce.WaitForMetric(ctx, logger.ToMainLog(), vm, metricName, 1*time.Hour, nil); err != nil {
-		return nonRetryable, err
+	// Check if expected_metrics.yaml exists, and run the test cases if it does.
+	testCaseBytes, err = readFileFromScriptsDir(path.Join("applications", app, "expected_metrics.yaml"))
+	if err == nil {
```

Comment:
(a) You only use `testCaseBytes` within the body of the conditional, so you can use:
```
	if testCaseBytes, err := readFileFromScriptsDir(path.Join("applications", app, "expected_metrics.yaml")); err == nil {
```
(and the same for the `expected_logs.yaml` check above).
But (b) it may make sense to invert this check instead and save yourself a level of indentation:
```
	testCaseBytes, err = readFileFromScriptsDir(path.Join("applications", app, "expected_metrics.yaml"))
	if err != nil {
		return nonRetryable, nil
	}
	logger.ToMainLog().Println("found expected_metrics.yaml, running metrics test cases...")
	err = runMetricsTestCases(ctx, logger, vm, testCaseBytes)
	return nonRetryable, err
```
(optionally inlining the last call, i.e., `return nonRetryable, runMetricsTestCases(ctx, logger, vm, testCaseBytes)`).

Reply:
My hesitation for (b) is that it only makes sense to apply that pattern to the final step in `runSingleTest`. Each time we add another step we'd have to un-refactor the previous step which is a bit annoying -- at least with (a) the steps are more-or-less copy-paste, so I'd prefer that direction.

Reply:
I landed on (a), which is pushed now.